tqdm/gui.py

```diff
@@ -316,6 +316,245 @@ def tgrange(*args, **kwargs):
     return tqdm_gui(_range(*args), **kwargs)
 
 
+class tqdm_tk(std_tqdm):
+    """
+    Experimental Tkinter GUI version of tqdm!
+    """
+
+    def __init__(self, *args, **kwargs):
+        try:
+            import tkinter
+            import tkinter.ttk as ttk
+        except ImportError:
+            import Tkinter as tkinter
+            import ttk as ttk
+        try:
+            grab = kwargs.pop("grab")
+        except KeyError:
+            grab = False
+        try:
+            tk_parent = kwargs.pop("tk_parent")
+        except KeyError:
+            tk_parent = None
+        try:
+            self._cancel_callback = kwargs.pop("cancel_callback")
+        except KeyError:
+            self._cancel_callback = None
+        try:
+            bar_format = kwargs.pop("bar_format")
+        except KeyError:
+            bar_format = None
+
+        kwargs["gui"] = True
+
+        # Tkinter specific default bar format
+        if bar_format is None:
+            kwargs["bar_format"] = (
+                "{n_fmt}/{total_fmt}, {rate_noinv_fmt}\n"
+                "{elapsed} elapsed, {remaining} ETA\n\n"
+                "{percentage:3.0f}%"
+            )
+
+        # This signals std_tqdm that it's a GUI but no need to crash
+        # Maybe there is a better way?
+        self.sp = object()
+
+        super(tqdm_tk, self).__init__(*args, **kwargs)
+
+        # Discover parent widget
+        if tk_parent is None:
+            # this will error if tkinter.NoDefaultRoot() called
+            try:
+                tkparent = tkinter._default_root
+            except AttributeError:
+                raise ValueError("tk_parent required when using NoDefaultRoot")
+            if tkparent is None:
+                # use new default root window as display
+                self.tk_window = tkinter.Tk()
+            else:
+                # some other windows already exist
+                self.tk_window = tkinter.Toplevel()
+        else:
+            self.tk_window = tkinter.Toplevel(tk_parent)
+
+        if self.disable:
+            return
+
+        warn('GUI is experimental/alpha', TqdmExperimentalWarning, stacklevel=2
+             )
+        self.tk_dispatching = self.tk_dispatching_helper()
+        if not self.tk_dispatching:
+            # leave is problematic if the mainloop is not running
+            self.leave = False
+
+        self.tk_window.protocol("WM_DELETE_WINDOW", self.cancel)
+        self.tk_window.wm_title("tqdm_tk")
```

Comment:
makes more sense to me to use

```suggestion
        self.tk_window.wm_title(self.desc)
```

and remove `self.tk_desc_var` etc...

Reply:
I've been using multi-line strings for `desc`, which doesn't work in the title bar. also serious users can reach in and set `wm_title` as well if they want!  

I guess the real question for you is: are multi-line description strings generally supported in `tqdm`? If yes, we have to stick with the `tk_desc_label` stuff.